src/impl_/pyclass.rs

```diff
@@ -1138,6 +1138,9 @@ pub trait PyClassBaseType: Sized {
     type BaseNativeType;
     type Initializer: PyObjectInit<Self>;
     type PyClassMutability: PyClassMutability;
+    /// Whether `__new__` ([ffi::PyTypeObject::tp_new]) for this type accepts arguments other
+    /// than the type of object to create.
+    const NEW_ACCEPTS_ARGUMENTS: bool = true;
```

Comment:
this was the neatest place I could find for this. Also considered making it a generic argument of `PyNativeTypeInitializer<T, NEW_ACCEPTS_ARGUMENTS>` but that seemed like a more invasive change.

This only allows arguments to be propagated or not, which is not as flexible as python where `super().__new__` is called manually so the class has full control. Should be good enough for most cases though?